### PR TITLE
DM-694 strip idpUsername suffix

### PR DESCRIPTION
**Pull request by sutherlanda**
This PR includes a migration to update existing usernames (needs to be tested in dev/test prior to production deployment as the migration would not be easily reversed).

Also includes modification to auth router to remove suffix on new user sign-up.

### CODE REVIEW

File: src/migrations/knexfile.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+});
 
 module.exports = {
   client: 'pg',
-  connection: POSTGRES_URL,
+  connection: process.env.POSTGRES_URL,
   migrations: {
-    tableName: DB_MIGRATIONS_TABLE_NAME,
+    tableName: 'migrations',
     directory: './tasks'
```
Comment (dhruvio) [inline src/migrations/knexfile.ts:14]: TODO: I need to implement support for building migrations to JS, and running the JS directly to avoid this change

**Comment by sutherlanda** [comment]
Added an extra check to confirm that when removing `@github` suffix the account is a Vendor type, and similarly for `@idir` and Government account types.  

For the data migration, also allow Admin account types for both `@github` and `@idir`.

**Review by dhruvio — COMMENTED** [review]
Looks good, I need to fix the migrations build system before we can merge

File: src/migrations/knexfile.ts (hunk 1 above)
Comment (dhruvio) [reply src/migrations/knexfile.ts]: So, I ended up figuring out the simpler solution is to properly resolve the REPOSITORY_ROOT_DIR variable in `back-end/config`. I am doing this by using the `find-up` package to find `package.json` and resolve its dirname. The build system stays the same for migrations.

**Pull request merged**